eng/Testing.targets

```diff
@@ -1,3 +1,96 @@
 <Project>
 
+  <!--
+      Environment variables:
+        - IsGitHubActionsRunner:       indicates whether tests are currently run on GitHub Actions; computed, overridable. Locally this can be set by "/p:GITHUB_ACTIONS=true".
+        - IsAzdoCIRunner:              indicates whether tests are currently run on Azure DevOps; computed, overridable. Locally this can be set by "/p:SYSTEM_TEAMPROJECT=foo".
+        - IsAzdoHelixRunner:           indicates whether tests are currently run on Helix; computed, overridable. Locally this can be set by "/p:PrepareForHelix=true".
+
+      Project capabilities:
+        - IsTestProject:               indicates whether the project is a test project; default is false; computed, overridable.
+        - IncludeTestUtilities:        indicates whether the test project must not include the TestUtilities project reference; default is false; overridable.
+
+      Project requirements:
+        - RunOnGithubActions:          indicates whether tests should run on GitHub Actions (either Windows or Linux); computed.
+        - RunOnGithubActionsWindows:   indicates whether tests should run on Windows in GitHub Actions; default is true; overridable.
+        - RunOnGithubActionsLinux:     indicates whether tests should run on Linux in GitHub Actions; default is true; overridable.
+        - RunOnAzdoCI:                 indicates whether tests should run on Azure DevOps (either Windows or Linux); always false, if RunOnAzdoHelix=true; computed.
+        - RunOnAzdoCIWindows:          indicates whether tests should run on Windows in Azure DevOps; default is true; overridable.
+        - RunOnAzdoCILinux:            indicates whether tests should run on Linux in Azure DevOps; default is true; overridable.
+        - RunOnAzdoHelix:              indicates whether tests should run on Helix (either Windows or Linux); computed.
+        - RunOnAzdoHelixWindows:       indicates whether tests should run on Windows in Helix; default is true; overridable.
+        - RunOnAzdoHelixLinux:         indicates whether tests should run on Linux in Helix; default is true; overridable.
+    -->
+
+  <PropertyGroup>
+    <!-- See https://docs.github.com/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#default-environment-variables -->
+    <IsGitHubActionsRunner Condition=" '$(GITHUB_ACTIONS)' == 'true' ">true</IsGitHubActionsRunner>
+
+    <!-- See https://learn.microsoft.com/azure/devops/pipelines/build/variables#system-variables -->
+    <IsAzdoCIRunner Condition=" '$(SYSTEM_TEAMPROJECT)' != '' ">true</IsAzdoCIRunner>
+
+    <IsAzdoHelixRunner Condition=" '$(PrepareForHelix)' == 'true' ">true</IsAzdoHelixRunner>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <RunOnGithubActions>false</RunOnGithubActions>
+    <RunOnGithubActions Condition=" '$(RunOnGithubActionsWindows)' == 'true' or '$(RunOnGithubActionsLinux)' == 'true' ">true</RunOnGithubActions>
```

Comment:
Is this logic correct?
Shouldn't `RunOnGithubActionsWindows` be considered only when on Windows, and `RunOnGithubActionsLinux` be considered only on Linux, instead of ORing them together without any OS checks?

Currently, it seems like Playground is run on GitHub Actions on Windows (VSTest just ignores when no tests are run, but with my PR for MTP I'm getting zero tests ran for Playground)

Reply:
Yes, the logic is correct.
These variables denote the "capabilities":
https://github.com/dotnet/aspire/blob/ac098f78e71602efa45f94bfb2918e0103e809a0/eng/Testing.targets#L13-L22

...and used to generate runsheets, which then executed on individual build agents.

You can see how it is being used in https://github.com/dotnet/aspire/pull/8618/files#diff-ce47bb41900ee3082587fce1184e86392a2c25a05882281792f04e8e8abc51f4.

Reply:
After a third look and two cups of tea later, I see the issue now. Thank you for pointing it out.
`SkipTests` should be OS-aware, so that we don't run a test on Windows, if the test project declared it couldn't be run on Windows. This is getting fixed in #8687.